### PR TITLE
Refactor method/member translation tests into their own suite

### LINKED ISSUE #34872: Have a tests suite (or model) specifically for method/member translations
When we want to add tests for method/member translations, we sometimes do that in Northwind, and sometimes in GearsOfWar. Northwind is generally bad because it has very few types, and has restricted data for those types (we can't add data to Northwind - it's frozen). GearsOfWar has the disadvantage of having lots of different versions (for different inheritance types, etc.), so if you add a small translation test, you suddenly have to add lots of versions of that tests - with duplicated SQL baselines - though that doesn't add any meaningful coverage. GearsOfWar also generally evolved to be a bit of a fallback/mess, where we frequently shove stuff into it without too much systematic design etc.; it's also not implemented by Cosmos because there's too much incompatible stuff in there.

We should consider having a test suite geared specifically for method/member translations. The model would be simple, no relationalships, and aim to cover all the basic database types and relevant data for all the types as well. We'd take care that tests there don't break as data as added, so that we can easily add data to support new tests without having to unbreak old ones. This would start us moving away from Northwind/GearsOfWar, in a more structured/clean direction etc.

If the number of tests is too big, we can consider having a common model with multiple test suites (like the Northwind tests).

### PR DESCRIPTION
**Pull request by roji**
This introduces a new BasicTypes model, and reorganizes many of our method/member translation tests into new test suites under Query/Translations.

* The tests have a simple model containing all basic, universally supported data. This is better than Northwind (which has only a subset of the types we need, fixed data, and cannot be updated) and GearsOfWar (which is generally a mess, is relational-only because of the use of navigations, and requires updating many times because of the different duplications).
* Tests are organized naturally based on the translation they exercise (e.g. TemporalTranslations, StringTranslations...) rather than the LINQ operator they're in (NorthwindWhereQueryTestBase, NorthwindSelectQueryTestBase). This makes it very easy to know where to find or put a test for a given translation.
* This is non-exhaustive: there are still many translation tests in Northwind/GearsOfWar/elsewhere that should move into the new tests. Doing this fully is a pretty huge task, but this is big step in that direction.
* Where we were testing both Where and Select, combined those into the same test (see #35318).
* Generally did lots of cleanup on all affected tests.
* All tests have SQL assertions, for all providers.
* All code has NRTs turned on.

Closes #34872

### CODE REVIEW

File: test/EFCore.Cosmos.FunctionalTests/Query/Translations/BasicTypesQueryCosmosFixture.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+public class BasicTypesQueryCosmosFixture : BasicTypesQueryFixtureBase
+{
+    protected override ITestStoreFactory TestStoreFactory => CosmosTestStoreFactory.Instance;
+
+    public TestSqlLoggerFactory TestSqlLoggerFactory
+        => (TestSqlLoggerFactory)ListLoggerFactory;
+
+    public override DbContextOptionsBuilder AddOptions(DbContextOptionsBuilder builder)
+        => builder.ConfigureWarnings(o => o.Ignore(CosmosEventId.NoPartitionKeyDefined));
```
Comment (AndriySvyryd) [inline test/EFCore.Cosmos.FunctionalTests/Query/Translations/BasicTypesQueryCosmosFixture.cs:16]: Ideally all Cosmos models should define partition keys

File: test/EFCore.Specification.Tests/TestModels/BasicTypesModel/BasicFlagsEnum.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+public enum BasicFlagsEnum
+{
+    Zero = 0,
+    One = 1,
+    Two = 2,
+    Four = 4,
+    Eight = 8,
+    Sixteen = 16,
+    ThirtyTwo = 32,
+}
```
Comment (AndriySvyryd) [inline test/EFCore.Specification.Tests/TestModels/BasicTypesModel/BasicFlagsEnum.cs:16]: Add composite and skips - 3, 128

File: test/EFCore.Specification.Tests/TestModels/BasicTypesModel/BasicTypesEntity.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    public DateTime DateTime { get; set; }
+    public DateOnly DateOnly { get; set; }
+    public TimeOnly TimeOnly { get; set; }
+    public DateTimeOffset DateTimeOffset { get; set; }
+    public TimeSpan TimeSpan { get; set; }
+
+    public bool Bool { get; set; }
+    public Guid Guid { get; set; }
+    public required byte[] ByteArray { get; set; }
```
Comment (AndriySvyryd) [inline test/EFCore.Specification.Tests/TestModels/BasicTypesModel/BasicTypesEntity.cs:28]: Missing unsigned types, char, Half 😁, Uri, IPAddress, PhysicalAddress

**Review by AndriySvyryd — APPROVED** [review]

File: test/EFCore.Specification.Tests/TestModels/BasicTypesModel/BasicTypesEntity.cs (hunk 3 above)
Comment (roji) [reply test/EFCore.Specification.Tests/TestModels/BasicTypesModel/BasicTypesEntity.cs:28]: We can add these as needed no? Especially given we don't actually support any translations e.g. on IPAddress/PhysicalAddress? For PG which supports these, I was planning to add properties to the model (e.g. in a separate entity) and have extra translations test suites to exercise that (something like this already in EFCore.PG).

File: test/EFCore.Cosmos.FunctionalTests/Query/Translations/BasicTypesQueryCosmosFixture.cs (hunk 1 above)
Comment (roji) [reply test/EFCore.Cosmos.FunctionalTests/Query/Translations/BasicTypesQueryCosmosFixture.cs:16]: OK, adding the Id as the partition key.

File: test/EFCore.Specification.Tests/TestModels/BasicTypesModel/BasicTypesEntity.cs (hunk 3 above)
Comment (roji) [reply test/EFCore.Specification.Tests/TestModels/BasicTypesModel/BasicTypesEntity.cs:28]: @AndriySvyryd I'm going to go ahead and merge this without the additional types, for now. I agree with you that at least some of these make sense, but let's add them later as needed (I've already spent enough time here...).

**Comment by ChrisJollyAU** [comment]
@roji Thanks for this change. It actually found a bug in my EFCore.Jet.

Turns out I was using the wrong truncate function (`Int` where I should be using `Fix`). It worked with Northwind because all the columns that we were doing a `Math.Truncate` on where positive values (`OrderId` and `UnitPrice`). The new tests had negative values in the data which had a different behaviour than expected

**Comment by roji** [comment]
That's really great to hear @ChrisJollyAU, thanks - we're trying to improve the rigor and breadth of the tests, and it's great to know that helps. FWIW we're working on other improved tests for 10, specifically tests around various types of relationships - hopefully these help as well.

**Pull request merged**